### ISSUE TITLE
Face details not saving

### ISSUE THREAD

**Issue by arcadium1990** [NicooPasPris/nicoo_charcreator#17]
Hello !
First of all, I really like the work that has been done thx !
But... I do have a problem, when i'm creating a character, all the face shape disappear. Is there any way of avoid this situation ? Is it supposed to be saved in the 'skin' section from the database ?

Edit:
The problem have been solved by switching to the v1.0.0 of Skinchanger instead of the 1.0.1. It also solved the saving issue i had on my database. Thanks you for you time Twiitchter ^^

**Comment by phiomet** [comment]
Same here

**Comment by Twiitchter** [comment]
Did you add the required edit to the skinchanger resource?

This one? 
```lua
local Face = {[1] = 'nose_width', [2] = 'nose_peak_hight', [3] = 'nose_peak_lenght', [4] = 'nose_bone_high', [5] = 'nose_peak_lovering', [6] = 'nose_bone_twist', [7] = 'eyebrown_high', [8] = 'eyebrown_forward', [9] = 'cheeks_bone_high', [10] = 'cheeks_bone_width', [11] = 'cheeks_width', [12] = 'eyes_openning', [13] = 'lips_thickness', [14] = 'jaw_bone_width', [15] = 'jaw_bone_back_lenght', [16] = 'chimp_bone_lowering', [17] = 'chimp_bone_lenght', [18] = 'chimp_bone_width', [19] = 'chimp_hole', [20] = 'neck_thickness'}
local HeadOverlay = {[0] = 'blemishes', [1] = 'beard', [2] = 'eyebrows', [3] = 'age', [4] = 'makeup', [5] = 'blush', [6] = 'complexion', [7] = 'sun', [8] = 'lipstick', [9] = 'moles', [10] = 'chest', [11] = 'bodyb'}
RegisterNetEvent('skinchanger:saveWithFiveMFunc')
AddEventHandler('skinchanger:saveWithFiveMFunc', function(index, type, v)
	local playerPed = PlayerPedId()
	if type == 'headoverlay' then
		Character[index..'_1'] = v[1]
		Character[index..'_2'] = v[2]
		if v[3] then
			Character[index..'_3'] = v[3]
		end
		if v[4] then
			Character[index..'_4'] = v[4]
		end
	elseif type == 'componentvariation' then
		if index == 'arms' then
			Character[index] = v[1]
			Character[index..'_2'] = v[2]
		elseif index == 'sex' then
			Character[index] = v[1]
		else
			Character[index..'_1'] = v[1]
			Character[index..'_2'] = v[2]
		end
	elseif type == 'face' then
		Character[index] = v
	elseif type == 'facefeature' then
		Character[Face[index]] = v
	end
end)
```

**Comment by arcadium1990** [comment]
Yes, I did put that above the function. Btw, the number of the line have changed, it was 223, maybe because of a skinchanger update.

**Comment by Twiitchter** [comment]
Can you please show me how the data looks from your Database?

**Comment by arcadium1990** [comment]
Sure ! ^^
![Sans titre](https://user-images.githubusercontent.com/69057221/89119966-817de480-d4b2-11ea-8f85-2e26f72b918a.png)

This is the result i have from the script in the skin column: {"eyebrows_5":0.0,"lipstick_2":0,"bodyb_2":0,"moles_1":0,"bproof_2":0,"sun_1":0,"chain_2":0,"hair_color_2":0,"arms_2":0,"beard_4":0,"sun_2":0,"sex":0,"nose_peak_hight":-0.28,"mask_1":0,"eyebrows_3":0,"makeup_2":0,"mom":21,"cheeks_2":0.0,"bracelets_2":0,"moles_2":0,"chest_2":0,"nose_5":0.0,"arms":2,"chest_3":0,"beard_2":55,"skin_mx":0.13,"chest_1":0,"shoes_2":9,"complexion_1":0,"tshirt_1":75,"age_1":0,"chin_height":0.0,"chimp_bone_width":0.0,"chain_1":0,"dad":15,"torso_2":0,"blemishes_2":0,"lips_thick":0.0,"hair_1":19,"tshirt_2":0,"blush_3":0,"lipstick_3":0,"helmet_2":0,"chin_lenght":0.0,"mask_2":0,"shoes_1":40,"face":0,"eyebrows_6":0.0,"glasses_2":0,"torso_1":72,"bags_1":0,"nose_bone_twist":0.23,"jaw_1":0.0,"bags_2":0,"pants_1":24,"skin":0,"hair_2":0,"model_hash":1885233650,"pants_2":0,"chin_hole":0.0,"chimp_bone_lenght":0.44,"helmet_1":-1,"lipstick_4":0,"jaw_bone_width":0.78,"glasses_1":-1,"decals_1":0,"eyebrows_2":0,"makeup_1":0,"ears_1":-1,"decals_2":0,"shape_mx":0.0,"nose_4":0.0,"chin_width":0.0,"makeup_3":0,"jaw_bone_back_lenght":0.92,"eye_open":0.0,"eye_color":0,"age_2":0,"chimp_bone_lowering":0.45,"nose_3":0.0,"blush_1":0,"makeup_4":0,"beard_3":0,"complexion_2":0,"eyebrows_4":0,"watches_1":-1,"cheeks_1":0.0,"nose_peak_lenght":0.42,"blemishes_1":0,"nose_1":0.0,"hair_color_1":0,"lipstick_1":0,"eyebrows_1":0,"neck_thick":0.0,"bodyb_1":0,"nose_6":0.0,"bracelets_1":-1,"neck_thickness":-0.04,"beard_1":11,"jaw_2":0.0,"ears_2":0,"bproof_1":0,"watches_2":0,"blush_2":0,"cheeks_3":0.0,"nose_bone_high":0.0,"nose_2":0.0}

**Comment by Twiitchter** [comment]
It seems like it isn't quite saving correctly, either that, or you have other things editing it. 

Eg, you have an entry named nose_4, it wouldn't be able to match that up to apply that to your character. Same with jaw_2 and cheeks_3.

They names of the entries here should be the default ones plus the shape_mx, skin_mx, and all the ones from the Face table. Like nose peak etc.

If they don't have the names as they are listed within the skinchanger, ie..

```lua
local Face = {[1] = 'nose_width', [2] = 'nose_peak_hight', [3] = 'nose_peak_lenght', [4] = 'nose_bone_high', [5] = 'nose_peak_lovering', [6] = 'nose_bone_twist', [7] = 'eyebrown_high', [8] = 'eyebrown_forward', [9] = 'cheeks_bone_high', [10] = 'cheeks_bone_width', [11] = 'cheeks_width', [12] = 'eyes_openning', [13] = 'lips_thickness', [14] = 'jaw_bone_width', [15] = 'jaw_bone_back_lenght', [16] = 'chimp_bone_lowering', [17] = 'chimp_bone_lenght', [18] = 'chimp_bone_width', [19] = 'chimp_hole', [20] = 'neck_thickness'}
```
This will cause some issues.

Also within the client.lua, if you rename ageing to age, and freckles to moles they will start to work.
Just remeber to modify the messages.lua to refer to the new names.

**Comment by arcadium1990** [comment]
The problem have been solved by switching to the v1.0.0 of Skinchanger instead of the 1.0.1. It also solved the saving issue i had on my database. Thanks you for you time Twiitchter ^^
